gateway/dataimpl/server_v1/errorhandler.go

```diff
@@ -594,3 +594,50 @@ func (e ErrorHandler) NewKeyTooLongStatus(key string) *status.Status {
 	st := status.New(codes.InvalidArgument, fmt.Sprintf("Document key '%s' is too long", key))
 	return st
 }
+
+func (e ErrorHandler) NewInvalidBucketCreationArgumentStatus(baseErr error, bucketName string) *status.Status {
+	st := status.New(
+		codes.InvalidArgument,
+		fmt.Sprintf("Invalid argument error(s) returned by server: unable to parse server error please check debug info"))
+	st = e.tryAttachStatusDetails(st, &epb.ResourceInfo{
+		ResourceType: "bucket",
+		ResourceName: bucketName,
+		Description:  "",
+	})
+	st = e.tryAttachExtraContext(st, baseErr)
+
+	// Details of which arguemnts were invalid is contained in the response body
```

Comment:
What does parsing the body actually gain us here? Could we not just always include the error body in the error status above?  Is a durability impossible error actually invalid args or should it be failed precondition?

Reply:
There were a couple of main reasons. Firstly if we just pass the error body without any parsing we have no way of knowing/controlling what the User will see on the output. Secondly there are changes in naming between stellar-gateway and gocbcorex that could leave to confusing messages to the User, e.g in the `durability_min_level` case addressed below.

Reply:
I agree that durability impossible should be failed precondition. Since a large part of the distinction is that an invalid argument will always be invalid, regardless of state, it seems to me that we should not be returning invalid argument statuses based on the response from the server and these should always be as the result of preliminary validation. If an argument is, and always will be invalid, then we should be able to know this without "asking" the server. The issue then becomes that we have multiple places we set valid values for fields, that could cause problems if we ever change them, e.g. if buckets are changed to allow less than 100MB RAM. However since these are pretty fundamental building blocks, is it safe to assume that such values won't change?